--- a/docopt_sh/docopt.sh	2019-09-29 21:24:12.144254907 +0300
+++ b/docopt_sh/docopt.sh	2019-09-29 21:24:29.197225937 +0300
@@ -74,7 +74,7 @@
 parse() {
   if ${DOCOPT_DOC_CHECK:-true}; then
     local doc_hash
-    doc_hash=$(printf "%s" "$DOC" | shasum -a 256)
+    doc_hash=$(printf "%s" "$DOC" | sha256sum)
     if [[ ${doc_hash:0:5} != "$digest" ]]; then
       stderr "The current usage doc (${doc_hash:0:5}) does not match \
 what the parser was generated with (${digest})
@@ -405,4 +405,3 @@
   printf -- "exit %d\n" "$1"
   exit "$1"
 }
-
